runner/statuses.go

```diff
@@ -0,0 +1,70 @@
+package runner
+
+import (
+	"time"
+)
+
+// StateMask describes a set of States as a bitmask.
+type StateMask uint64
+
+// StatusQuery describes a query for Statuses.
+// The Runs and States are and'ed: a ProcessStatus matches a StatusQuery
+// if its ID is in q.Runs (or q.AllRuns) and its state is in q.States
+type StatusQuery struct {
+	Runs    []RunId   // Runs to query for
+	AllRuns bool      // Whether to match all runs
+	States  StateMask // What States to match
+}
+
+// Statuses allows Querying the status of Processes
+type Statuses interface {
+	// QueryWait queries the statuses for statuses that match q, waiting if none match now.
+	// Returns the ProcessStatus'es that match the Query.
+	// (i.e., if the query is for 3 RunIds but only one matches the StateMask,
+	// only one will be returned)
+	// If timeout is 0, same as QuerySync
+	// If positive, wait up to that long
+	Query(q StatusQuery, timeout time.Duration) ([]ProcessStatus, error)
```

Comment:
I dont think we should attempt to create a unifying abstraction around getting the status of a local call vs a remote call, at least right now

These are fundamentally different and doing this is not trivial nor a good idea see ([A Note on Distributed Computing](http://citeseerx.ist.psu.edu/viewdoc/summary?doi=10.1.1.41.7628))